File: appinventor/components-ios/src/NotifierLength.swift
// -*- mode: swift; swift-mode:basic-offset: 2; -*-
// Copyright 2026 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// https://www.apache.org/licenses/LICENSE-2.0

@objc public class NotifierLength: NSObject, OptionList {
  @objc public static let Short = NotifierLength(0)
  @objc public static let Long = NotifierLength(1)

  private static let LOOKUP: [Int32: NotifierLength] = generateOptionsLookup(Short, Long)

  let value: Int32

  @objc private init(_ value: Int32) {
    self.value = value
  }

  @objc public func toUnderlyingValue() -> AnyObject {
    return value as AnyObject
  }

  @objc public static func fromUnderlyingValue(_ value: Int32) -> NotifierLength? {
    return LOOKUP[value]
  }
}

File: appinventor/components/src/com/google/appinventor/components/common/NotifierLength.java
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright 2026 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

package com.google.appinventor.components.common;

import java.util.HashMap;
import java.util.Map;

/**
 * Defines how long Notifier alerts should stay visible.
 */
public enum NotifierLength implements OptionList<Integer> {
  Short(0),
  @Default
  Long(1);

  private final int value;

  NotifierLength(int value) {
    this.value = value;
  }

  public Integer toUnderlyingValue() {
    return value;
  }

  private static final Map<Integer, NotifierLength> lookup = new HashMap<>();

  static {
    for (NotifierLength length : NotifierLength.values()) {
      lookup.put(length.toUnderlyingValue(), length);
    }
  }

  public static NotifierLength fromUnderlyingValue(Integer length) {
    return lookup.get(length);
  }
}
